### PR TITLE
More robust retrieval for PostgreSQL version

### LINKED ISSUE #54379: DB Manager: bad PostgreSQL version parsing for beta versions
### What is the bug or the crash?

When using a PostgreSQL beta version (for instance PG16beta3), DB manager crashes when trying to get version number as the parsed value is not an int (16beta3 for instance).

File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins/db_manager/db_plugins/postgis/connector.py", line 651, in getTableFields
    version_number = int(self.getInfo()[0].split(' ')[1].split('.')[0])
ValueError: invalid literal for int() with base 10: '16beta3'

### Steps to reproduce the issue

Configure a PostgreSQL beta version connection in postgis
Use DB manager to load a table
Crash occurs.

### Versions

3.30.0-'s-Hertogenbosch - QGIS code revision f186b8efe0

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nicogodet**
## Description

- Fixes #54379

https://regex101.com/r/tEX6YB/4

### CODE REVIEW

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 1]
```diff
@@ -348,6 +348,13 @@ def getInfo(self):
         self._close_cursor(c)
         return res
 
+    def getPsqlVersion(self):
+        regex = r"^PostgreSQL\s([0-9]{1,2})"
+        match = re.match(regex, self.getInfo()[0])
+        if match:
+            return int(match.group(1))
+        return
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/postgis/connector.py:356]: Should this give a meaningful error if no match?

Comment (nicogodet) [reply python/plugins/db_manager/db_plugins/postgis/connector.py:356]: Ideally, it should.
But I looked at other functions and none of them raise an exception.

https://github.com/qgis/QGIS/blob/5d58e8ac917cf8201faba63686cc81cd9c6fc300/python/plugins/db_manager/db_plugins/postgis/connector.py#L351-L369

I replaced latest `return` with `raise DbError("Unrecognized PostgreSQL version")` and this error is raised quite late when using DB Manager (only when showing layer info and not at connection).
image

I can commit this change if you want but all other functions should be updated and it's not in my plan to do it.

Comment (m-kuhn) [reply python/plugins/db_manager/db_plugins/postgis/connector.py:356]: I think that's better than showing it nowhere?
I would add the unparsable string to the error message to make it easier fixable if raised.

Comment (nicogodet) [reply python/plugins/db_manager/db_plugins/postgis/connector.py]: Done in https://github.com/qgis/QGIS/pull/54401/commits/8fb3352348d810aea73f16e77d807acd2f8a8b3f

**Review by m-kuhn — APPROVED** [review]

**Comment by nicogodet** [comment]
unrelated MingW failure

**Comment by agiudiceandrea** [comment]
It seems to me the issue also occur in 3.32 and 3.28 and thus this PR needs to be backported to the related branches. What do you think @nicogodet @m-kuhn.

**Comment by nicogodet** [comment]
I wasn't confident enough with first try but with the raised error, OK for me to backport to 3.32 and pending 3.28.

**Comment by nicogodet** [comment]
Currently, no crash, "just" python error which make DB Manager unusable but QGIS still works

**Pull request merged**